pkg/checks/checks.go

```diff
@@ -71,39 +71,122 @@ func checkNamespace(namespace string) bool {
+		Namespace: "default",
+	},
+	{
+		Name:      "router",
+		Namespace: "default",
+	},
+	{
+		Name:      "registry-console",
+		Namespace: "default",
+	},
+	{
+		Name:      "apiserver",
+		Namespace: "kube-service-catalog",
+	},
+	{
+		Name:      "controller-manager",
+		Namespace: "kube-service-catalog",
+	},
+	{
+		Name:      "bootstrap-autoapprover",
+		Namespace: "openshift-infra",
+	},
+	{
+		Name:      "asb",
+		Namespace: "openshift-ansible-service-broker",
+	},
+	{
+		Name:      "apiserver",
+		Namespace: "openshift-template-service-broker",
+	},
+	{
+		Name:      "bootstrap-autoapprover",
+		Namespace: "openshift-infra",
+	},
+	{
+		Name:      "webconsole",
+		Namespace: "openshift-web-console",
+	},
+}
+var staticWhitelist = []struct {
+	Name      string
+	Namespace string
+	NodeType  string
+}{
+	{
+		Name:      "api",
+		Namespace: "kube-system",
+		NodeType:  "master",
+	},
+	{
+		Name:      "controllers",
+		Namespace: "kube-system",
+		NodeType:  "master",
+	},
+	{
+		Name:      "etcd",
+		Namespace: "kube-system",
+		NodeType:  "master",
+	},
+	{
+		Name:      "logbridge",
+		Namespace: "kube-system",
+		NodeType:  "all",
+	},
+	{
+		Name:      "sync-master-000000",
+		Namespace: "kube-system",
+	},
+}
+var daemonsetWhitelist = []struct {
+	Name      string
+	Namespace string
+}{
+	{
+		Name:      "prometheus-node-exporter",
+		Namespace: "openshift-metrics",
+	},
+	{
+		Name:      "sync",
+		Namespace: "openshift-node",
+	},
+	{
+		Name:      "ovs",
+		Namespace: "openshift-sdn",
+	},
+	{
+		Name:      "sdn",
+		Namespace: "openshift-sdn",
+	},
+}
+var statefulsetWhitelist = []struct {
+	Name      string
+	Namespace string
+	NodeType  string
+}{
+	{
+		Name:      "prometheus",
+		Namespace: "openshift-metrics",
+	},
+}
+
 // WaitForInfraServices verify daemonsets, statefulsets
-func WaitForInfraServices(ctx context.Context, appclient *appclient.AppsV1Client) error {
+func WaitForInfraServices(ctx context.Context, kc *kubernetes.Clientset) error {
 	log.Info("checking infrastructure service health")
-out:
-	for {
-		_, err := appclient.Deployments("openshift-web-console").Get("webconsole", metav1.GetOptions{})
-		switch {
-		case err == nil:
-			break out
-		case kerrors.IsNotFound(err):
-		default:
-			return err
-		}
-		select {
-		case <-time.After(2 * time.Second):
-		case <-ctx.Done():
-			return ctx.Err()
-		}
-	}
-
-	// Daemonsets
-	dsList, err := appclient.DaemonSets("").List(metav1.ListOptions{})
-	if err != nil {
-		return err
-	}
 
-	for _, ds := range dsList.Items {
-		if !checkNamespace(ds.Namespace) {
-			continue
-		}
-		log.Info(fmt.Sprintf("checking %s %q", ds.Kind, ds.Name))
+	for idx, app := range daemonsetWhitelist {
+		log.Info(fmt.Sprintf("checking   daemonset[%d]: [%s] %s", idx, app.Namespace, app.Name))
 		for {
-			ds, err := appclient.DaemonSets(ds.Namespace).Get(ds.Name, metav1.GetOptions{})
+			ds, err := kc.AppsV1().DaemonSets(app.Namespace).Get(app.Name, metav1.GetOptions{})
```

Comment:
are you sure you won't need to handle kerrors.IsNotFound below?

Reply:
You were correct. Added.